### PR TITLE
hardening/893_add_documentation_for_admin_methods

### PR DESCRIPTION
**Pull request by pcoello25**
* Partially fix issue #893 
  * Update CHANGES_NEXT_RELEASE is not neccesary.
* No tests are required
* Assignee: @frbattid

### CODE REVIEW

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 1]
```diff
@@ -290,8 +296,253 @@ Responses:
 
 [Top](#top)
 
-##<a name="section11"></a>`POST /v1/subscriptions`
-###<a name="section11.1"></a> `NGSI Version 1`
+##<a name="section11"></a>`GET /admin/configuration/agent`
+###<a name="section11.1"></a>`GET` all parameters
+
+Gets all the parameters from an agent adding the path to it in the request. The name of the agent must start with `agent_`.
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... given the path to the configuration file as the URI within the URL.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 2]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+GET "http://<cygnus_host>:<management_port>/admin/configuration/agent/cygnus/apache-flume-1.4.0-bin/conf/agent_cygnus.conf"
+```
+
+There are a second option for `GET` all the parameters, using `/v1/admin/configuration/agent`. Both have the same behaviour, you are free to use either of them.
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: NOTE: Using the `/v1/admin/configuration/agent` path behaves the same way.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 3]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+GET "http://<cygnus_host>:<management_port>/v1/admin/configuration/agent/cygnus/apache-flume-1.4.0-bin/conf/agent_cygnus.conf"
+```
+
+Responses:
+
+Valid path to the agent:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... to the agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 4]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+```
+
+Invalid path to the agent:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... to the agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 5]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+{"success":"false","result" : { "File not found in the path received" }
+```
+
+Invalid agent file name:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 6]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+{"success":"false","error":"Agent file name must start with 'agent_'."}
+```
+
+[Top](#top)
+
+###<a name="section11.2"></a>`GET` a single parameter
+
+Gets a single parameter from an agent adding the path to it in the request and the name of the parameter as a query parameter. The name of the agent must start with `agent_`.
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Same comment than above

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 7]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+GET "http://<cygnus_host>:<management_port>/admin/configuration/agent/cygnus/apache-flume-1.4.0-bin/conf/agent_cygnus.conf&param=<param_name>"
+```
+
+There are a second option for `GET` a single parameter, using `/v1/admin/configuration/agent`. Both have the same behaviour, you are free to use either of them.
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Same comment than above

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 8]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+GET "http://<cygnus_host>:<management_port>/v1/admin/configuration/agent/cygnus/apache-flume-1.4.0-bin/conf/agent_cygnus.conf&param=<param_name>"
+```
+
+Responses:
+
+Valid path to the agent:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... to the agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 9]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+{"success":"true","result" : {"cygnusagent.sinks.mysql-sink.data_model":"dm-by-attribute"}
+```
+
+Invalid path to the agent:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... to the agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 10]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+{"success":"false","result" : {"File not found in the path received"}
+```
+
+Parameter not found in the agent:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... in the agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 11]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+{"success":"false","result" : {"Param 'cygnusagent.sinks.mysql-sink.new_parameter' not found in the agent"}
+```
+
+Invalid agent file name:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 12]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+{"success":"false","error":"Agent file name must start with 'agent_'."}
+```
+
+[Top](#top)
+
+##<a name="section12"></a>`POST /admin/configuration/agent`
+
+Posts a single parameter if it doesn't exist in the agent, adding the path to it in the request and the name and the value of the parameter as a query parameters. The name of the agent must start with `agent_`.
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Same comment than above

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 13]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+POST "http://<cygnus_host>:<management_port>/admin/configuration/agent/cygnus/apache-flume-1.4.0-bin/conf/agent_cygnus.conf?param=cygnusagent.sinks.mysql-sink.my_new_param&value=my_new_value"
+```
+
+There are a second option for `GET` a single parameter, using `/v1/admin/configuration/agent`. Both have the same behaviour, you are free to use either of them.
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Same comment than above

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 14]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+POST "http://<cygnus_host>:<management_port>/v1/admin/configuration/agent/cygnus/apache-flume-1.4.0-bin/conf/agent_cygnus.conf?param=cygnusagent.sinks.mysql-sink.my_new_param&value=my_new_value"
+```
+
+Responses:
+
+Valid path to the agent:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... to the agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 15]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+```
+
+Invalid path to the agent:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... to the agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 16]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+```
+
+Invalid agent file name:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 17]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+{"success":"false","error":"Agent file name must start with 'agent_'."}
+```
+
+[Top](#top)
+
+##<a name="section13"></a>`PUT /admin/configuration/agent`
+
+Puts a single parameter if it doesn't exist or update it if already exists in the agent, adding the path to it in the request and the name and the value of the parameter as a query parameters. The name of the agent must start with `agent_`.
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Same comment than above

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 18]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+PUT "http://<cygnus_host>:<management_port>/admin/configuration/agent/cygnus/apache-flume-1.4.0-bin/conf/agent_cygnus.conf?param=cygnusagent.sinks.mysql-sink.my_new_param&value=my_new_value"
+```
+
+There are a second option for `GET` a single parameter, using `/v1/admin/configuration/agent`. Both have the same behaviour, you are free to use either of them.
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Same comment than above

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 19]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+PUT "http://<cygnus_host>:<management_port>/v1/admin/configuration/agent/cygnus/apache-flume-1.4.0-bin/conf/agent_cygnus.conf?param=cygnusagent.sinks.mysql-sink.new_param&value=new_value"
+```
+
+Responses:
+
+Valid path to the agent:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... to the agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 20]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+```
+
+Invalid path to the agent:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... to the agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 21]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+{"success":"false","result" : { "File not found in the path received" }
+```
+
+Invalid agent file name:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 22]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+cygnusagent.sinks.mysql-sink.new_parameter = new_value
+cygnusagent.sinks.mysql-sink.data_model = dm-by-attribute
+cygnusagent.sinks.mysql-sink.attr_persistence = row
+```
+
+[Top](#top)
+
+##<a name="section14"></a>`DELETE /admin/configuration/agent`
+
+Deletes a single parameter if it exists in the agent, adding the path to it in the request and the name of the parameter as a query parameter. The name of the agent must start with `agent_`.
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Same comment than above

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 23]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+DELETE "http://<cygnus_host>:<management_port>/admin/configuration/agent/cygnus/apache-flume-1.4.0-bin/conf/agent_cygnus.conf?param=cygnusagent.sinks.mysql-sink.new_parameter
+```
+
+There are a second option for `DELETE` a single parameter, using `/v1/admin/configuration/agent`. Both have the same behaviour, you are free to use either of them.
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Same comment than above

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 24]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+
+```
+DELETE "http://<cygnus_host>:<management_port>/v1/admin/configuration/agent/cygnus/apache-flume-1.4.0-bin/conf/agent_cygnus.conf?param=cygnusagent.sinks.mysql-sink.new_param"
+```
+
+Responses:
+
+Valid path to the agent:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... to the agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 25]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+```
+
+Invalid path to the agent:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... to the agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md [hunk 26]
```diff
@@ -290,8 +296,253 @@ Responses:
...
+```
+
+Invalid agent file name:
```
Comment (frbattid) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: ... agent configuration file.

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 1 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 2 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 3 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 4 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 5 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 6 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 7 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 8 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 9 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 10 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 11 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 12 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 13 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 14 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 15 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 16 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 17 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 18 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 19 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 20 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 21 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 22 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 23 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 24 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 25 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

File: doc/cygnus-common/installation_and_administration_guide/management_interface.md (hunk 26 above)
Comment (pcoello25) [inline doc/cygnus-common/installation_and_administration_guide/management_interface.md]: Fixed in 5d935cf

**Comment by frbattid** [comment]
LGTM

**Pull request merged**